pkg/ccl/importccl/exportcsv_test.go

```diff
@@ -333,38 +336,70 @@ func validateParquetFile(t *testing.T, file string, test parquetTest) error {
 			t.Logf("\n Record %v:", count)
 			for i := 0; i < len(cols); i++ {
 				if test.vals[count][i] == nil {
-					// If we expect a null value, the row created by the parquet reader will not have the
-					// associated column.
+					// If we expect a null value, the row created by the parquet reader
+					// will not have the associated column.
 					_, ok := row[cols[i].SchemaElement.Name]
 					require.Equal(t, ok, false)
 					continue
 				}
-				var decodedV interface{}
 				v := row[cols[i].SchemaElement.Name]
-				switch vv := v.(type) {
-				case []byte:
-					// the parquet exporter encodes native go strings as []byte, so an extra
-					// step is required here
-					// TODO (MB): as we add more type support, this
-					// test will be insufficient: many go native types are encoded as
-					// []byte, so in the future, each column will have to call it's own
-					// custom decoder ( this is how IMPORT Parquet will work)
-					decodedV = string(vv)
-				case int64, float64, bool:
-					decodedV = vv
-				default:
-					t.Fatalf("unexepected type: %T", vv)
-				}
+				decodedV := decodeEl(t, v)
 				t.Logf("\t %v", decodedV)
 				require.Equal(t, test.vals[count][i], decodedV)
 			}
 			count++
 		}
 	}
-
 	return nil
 }
 
+func decodeEl(t *testing.T, v interface{}) interface{} {
+	var decodedV interface{}
+	switch vv := v.(type) {
+	case []byte:
+		// The parquet exporter encodes native go strings as []byte, so an extra
+		// step is required here.
+
+		// TODO (MB): as we add more type support, this switch statement will be
+		// insufficient: many go native types are encoded as []byte, so in the
+		// future, each column will have to call it's own custom decoder ( this is
+		// how IMPORT Parquet will work).
+		decodedV = string(vv)
+	case int64, float64, bool:
+		decodedV = vv
+	case map[string]interface{}:
+		var arrDecodedV []interface{}
+		b := vv["list"].([]map[string]interface{})
+
+		// Array values are stored in an array of maps: []map[string]interface{},
+		// where the ith map contains a single key value pair. The key is always "element"
+		// and the value is the ith value in the array.
+
+		// If the array of maps only contains an empty map, the array is empty. This
+		// occurs IFF "element" is not in the map.
+
+		// NB: there's a bug in the fraugster-parquet vendor library around reading
+		// an ARRAY[NULL]. See the "arrays" test case for more info. Ideally, once
+		// the bug gets fixed, ARRAY[NULL] will get read as the kvp {"element":interface{}} while
+		// ARRAY[] will continue to get read as an empty map.
+		if _, nonEmpty := b[0]["element"]; !nonEmpty {
+			arrDecodedV = []interface{}{}
```

Comment:
@stevendanna I think it makes more sense to decode ARRAY[] to []interface{}{}, instead of []interface{}(nil). Wanted to make sure you took a close look at this, as this goes against common go convention: https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices

Reply:
Makes sense to me.